### PR TITLE
Upgrade Go dependencies

### PR DESCRIPTION
**Pull request by cnunciato**
Re-runs the generator script, updates all pulumi/pulumi deps to the latest version, and clears out some unneeded go.mod entries.

### CODE REVIEW

File: serverless-gcp-go/go.mod [hunk 1]
```diff
@@ -7,86 +7,3 @@ require (
 	github.com/pulumi/pulumi-synced-folder/sdk v0.0.9
 	github.com/pulumi/pulumi/sdk/v3 v3.91.1
```
Comment (scottslowe) [inline serverless-gcp-go/go.mod:8]:
```suggestion
	github.com/pulumi/pulumi/sdk/v3 v3.96.1
```

File: static-website-gcp-go/go.mod [hunk 2]
```diff
@@ -7,83 +7,3 @@ require (
 	github.com/pulumi/pulumi-synced-folder/sdk v0.0.9
 	github.com/pulumi/pulumi/sdk/v3 v3.91.1
```
Comment (scottslowe) [inline static-website-gcp-go/go.mod:8]:
```suggestion
	github.com/pulumi/pulumi/sdk/v3 v3.96.1
```

File: vm-gcp-go/go.mod [hunk 3]
```diff
@@ -6,83 +6,3 @@ require (
 	github.com/pulumi/pulumi-gcp/sdk/v7 v7.0.0
 	github.com/pulumi/pulumi/sdk/v3 v3.91.1
```
Comment (scottslowe) [inline vm-gcp-go/go.mod:7]:
```suggestion
	github.com/pulumi/pulumi/sdk/v3 v3.96.1
```

**Review by scottslowe — CHANGES_REQUESTED** [review]
Looks like the generator script missed a couple templates, so the changes fix/update those. Otherwise LGTM.

**Comment by cnunciato** [comment]
Applied -- thanks @scottslowe!

**Review by scottslowe — APPROVED** [review]

**Pull request merged**